### PR TITLE
Principal axis factor analysis and rotation

### PR DESCRIPTION
**Pull request by yl565**
Continuation of #3294

### CODE REVIEW

**Comment by josef-pkt** [comment]
rebase looks good from what I can see on github

File: statsmodels/multivariate/factor.py [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+        communality = pd.DataFrame([self.communality],
+                                   columns=self.endog_names, index=[''])
+        summ.add_dict({'': ''})
+        summ.add_dict({'': 'Communality'})
+        summ.add_df(communality)
+        summ.add_dict({'': ''})
+        summ.add_dict({'': 'Pre-rotated loadings'})
+        summ.add_df(loadings_no_rot)
+        summ.add_dict({'': ''})
+        if self.rotation is not None:
```
Comment (josef-pkt) [inline statsmodels/multivariate/factor.py]: BUG from renamings
self.rotation -> self.rotation method
also on line 318

summary() needs a smoke test.
If the result is stable and doesn't change on repeated runs, then the full summary string could be tested as equal to a saved summary.

**Comment by josef-pkt** [comment]
@yl565  Can you add the R commands for producing the reference numbers in the unit tests to the test module?

I'm trying out an example from Steiger (using factanal and fa.promax) http://www.statpower.net/Content/312/Handout/Exploratory%20Factor%20Analysis%20with%20R.pdf
but the loadings after rotation don't match up at two decimals, they are still close except after promax the last factor has flipped sign, all loadings negative.

The results look sensitive to algorithmic details, and it's better to have the record for how the reference numbers were created, and which R function and which options correspond to our solution.

**Comment by yl565** [comment]
I added the R code in docstrings. I have to flip the sign on some of the loading vectors to be able to match the results though.

**Comment by josef-pkt** [comment]
Can you amend (and force push) the last commit to change the docstring of the unit test function to code comments with `#`?
nosetest was using the docstring in the unit test results (verbose and failure) instead of the function or method name. I guess pytest is doing something similar. That's why we never use a docstring for any `test_xxx` functions and methods.

**Comment by yl565** [comment]
OK, I amended the last comit

**Comment by josef-pkt** [comment]
all green, I merge this but still needs followup (api, rst, check docstring rendering, ...)

It's a good first part of factor analysis and includes rotation.

@yl565 Thank you

**Comment by yl565** [comment]
Thanks @josef-pkt

**Pull request merged**